compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -530,255 +536,102 @@ this = mkVar "this"
-          ERecCon _ [(_, tplSuperDict), _, _, _] <- convertExpr env x
-          tplTyLf@(TCon tplTyConLf) <- convertType env tplTy
-          choiceArgTyLf@(TConApp choiceTyCon _) <- convertType env choiceArgTy
-          let choiceName = ChoiceName (T.intercalate "." $ unTypeConName $ qualObject choiceTyCon)
-          choiceRetTyLf <- convertType env choiceRetTy
-          let exercise =
-                ETmLam (this, TContractId tplTyLf) $
-                  ETmLam (arg, choiceArgTyLf) $
-                    EUpdate $ UExercise
-                      tplTyConLf
-                      choiceName
-                      (EVar this)
-                      Nothing
-                      (EVar arg)
-          let toAnyChoice =
-                if envLfVersion env `supports` featureAnyType
-                  then
-                    ETyLam (proxyVar, KArrow KStar KStar) $
-                      ETmLam
-                        (mkVar "_", mkProxy tplTyLf) $
-                        ETmLam (mkVar "arg", choiceArgTyLf) $
-                          EToAny choiceArgTyLf (EVar arg)
-                  else EBuiltin BEError `ETyApp`
-                       TForall (proxyVar, KArrow KStar KStar) (mkProxy tplTyLf :-> choiceArgTyLf :-> TUnit) `ETmApp`
-                       EBuiltin (BEText "toAnyChoice is not supported in this DAML-LF version")
-          let fromAnyChoice =
-                if envLfVersion env `supports` featureAnyType
-                  then
-                    ETyLam (proxyVar, KArrow KStar KStar) $
-                      ETmLam
-                        (mkVar "_", mkProxy tplTyLf) $
-                        ETmLam (mkVar "any", TAny) $
-                          EFromAny choiceArgTyLf (EVar $ mkVar "any")
-                  else EBuiltin BEError `ETyApp`
-                         TForall (proxyVar, KArrow KStar KStar) (mkProxy tplTyLf :-> TUnit :-> TOptional choiceArgTyLf) `ETmApp`
-                         EBuiltin (BEText "fromAnyChoice is not supported in this DAML-LF version")
-          let dict = ERecCon (TypeConApp choiceTyLf [tplTyLf, choiceArgTyLf, choiceRetTyLf]) $
-                map (second (ETmLam (mkVar "_", TUnit)))
-                [ (mkIndexedField 1, tplSuperDict)
-                , (mkIndexedField 2, exercise)
-                , (mkIndexedField 3, toAnyChoice)
-                , (mkIndexedField 4, fromAnyChoice)
-                ]
-          name' <- convValWithType env name
-          pure [defValue name name' dict]
     -- This is inlined in the choice in the template so we can just drop this.
-    | "_choice_" `T.isPrefixOf` getOccText name = pure []
+    | "_choice_" `T.isPrefixOf` getOccText name
+    = pure []
+
+    -- Remove internal functions.
     | Just internals <- lookupUFM internalFunctions (envGHCModuleName env)
     , getOccFS name `elementOfUniqSet` internals
     = pure []
-    -- NOTE(MH): Desugaring `template X` will result in a type class
```

Comment:
I guess this is no longer necessary since we don’t have any default implementations?

Reply:
Right, and we don't want to filter out the `$c` methods either. We don't actually want templates to be special in any way with respect to GHC's & damlc's handling of typeclasses.